tests/pr_test.go

```diff
@@ -2,32 +2,59 @@
 package test
 
 import (
+	"github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper/common"
+	"log"
+	"os"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper/testhelper"
 )
 
-// Use existing resource group
-const resourceGroup = "geretain-test-postgres"
 const defaultExampleTerraformDir = "examples/default"
 const autoscaleExampleTerraformDir = "examples/autoscale"
 const fsCloudTerraformDir = "examples/fscloud"
 const completeExampleTerraformDir = "examples/complete"
 const replicaExampleTerraformDir = "examples/replica"
 
-// Restricting due to limited availability of BYOK in certain regions
-const regionSelectionPath = "../common-dev-assets/common-go-assets/icd-region-prefs.yaml"
+// Use existing resource group
+// Allow the tests to create a unique resource group for every test to ensure tests do not clash. This is due to the fact that the auth policy created by this module has to be scoped to the resource group and hence would clash if tests used same resource group.
+//const resourceGroup = "geretain-test-postgres"
 
-func TestRunDefaultExample(t *testing.T) {
-	t.Parallel()
+// Define a struct with fields that match the structure of the YAML data
+const yamlLocation = "../common-dev-assets/common-go-assets/common-permanent-resources.yaml"
 
+// Not all regions provide cross region support so value must be hardcoded https://cloud.ibm.com/docs/hs-crypto?topic=hs-crypto-regions&interface=ui.
```

Comment:
why are you saying cross region support? This is a COS bucket term?
The reason we are locking into us-south is because that is where our Hyper Protect instance is provisioned so the instance has to be provisioned in the same region

Reply:
Only the FSCloud test need to use us-south. Can we add back the code to allow the other tests to use a region in 

`const regionSelectionPath = "../common-dev-assets/common-go-assets/icd-region-prefs.yaml"`